fabric-rendering-fluids-v1/src/main/java/net/fabricmc/fabric/api/client/render/fluid/v1/OpaqueFluid.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.render.fluid.v1;
+
+import net.minecraft.client.render.BackgroundRenderer;
+import net.minecraft.client.render.FogShape;
+import net.minecraft.client.world.ClientWorld;
+import net.minecraft.entity.Entity;
+
+/**
+ * Defines a non-transparent fluid with a fog.
+ */
+public interface OpaqueFluid {
```

Comment:
Implementing a client-only interface on common classes is not really Fabric API-like design and it's annoying to spam `@Environment` annotations everywhere to prevent crashes on dedi servers. Probably some sort of registry would be better (maybe extending the existing fluid render handler thing with the fog?)

Reply:
Do you mean a sort of setting object to pass into the FluidRenderHandlerRegistry.INSTANCE.register method?

Like this?

```java
FluidRenderHandlerRegistry.INSTANCE.register(TestFluids.NO_OVERLAY, TestFluids.NO_OVERLAY_FLOWING, new SimpleFluidRenderHandler(
  new Identifier("fabric-rendering-fluids-v1-testmod:block/test_fluid_still"),
  new Identifier("fabric-rendering-fluids-v1-testmod:block/test_fluid_flowing"),
  0xFF5555
),
new FogParameters(.....)
);
```

Reply:
This could be passed inside a setting object inside the FlowableFluid class, but FlowableFluid has no settings, and the constructor is blank, so a class extending FlowableFluid must be created.
I wanted to avoid creating a new "fluid-api" non-client side module for a single class.

Reply:
I've fixed this issue, please check.
If possible, add a label "new-module", please.